src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -618,8 +909,14 @@ SPECTRE_ALWAYS_INLINE auto operator-(
   static_assert(
       tmpl::equal_members<op1_generic_indices, op2_generic_indices>::value,
       "The generic indices when subtracting two tensors must be equal. This "
-      "error occurs from expressions like R(ti_a, ti_b) - S(ti_c, ti_a)");
-  return TensorExpressions::AddSub<T1, T2, Args1, Args2, -1>(~t1, ~t2);
+      "error "
+      "occurs from expressions like R(ti_a, ti_b) - S(ti_c, ti_a)");
+  if constexpr (T2::num_ops_subtree <=
+                TensorExpressions::detail::max_num_ops_in_sub_expression<X2>) {
+    return TensorExpressions::AddSub<T1, T2, Args1, Args2, -1>(~t1, ~t2);
+  } else {
+    return -t2 + t1;
```

Comment:
The idea with this is that if `t2` is a TE subtree that contains more than the max # of ops, then move t2 to be the left operand so we can split it up since splitting would be advantageous given that it has "enough" operations to make it worth it. If it's the right operand, it can't be split, as the implementation in this PR only splits along the "leftmost" path from the root to the leftmost leaf. Let me know if you think this is a reasonable thing to do, or not!

Reply:
Note, however, that while this is beneficial if this swap ends up putting `t2` on this leftmost path, if this swap occurs somewhere off of the leftmost path (i.e. somewhere that won't be split up), then this swap will simply be adding a negation, with no benefit. So, perhaps, it is best to not do any switching if and until I implement splitting elsewhere in the tree, because I don't think it's possible for `operator-()` to know where exactly it's operands exist in the tree (i.e. where in arithmetic expression they exist)

Reply:
I'm not sure what advice to give you in terms of performance other than measure :) Feel free to leave the PR as-is and measure later! But I think documenting why it's being swapped, or pointing to where it's discussed in other documentation would be helpful here :)

Reply:
I decided to just remove the swap for now in new fixup https://github.com/sxs-collaboration/spectre/pull/3842/commits/904587939fe39bf688e0670b7b21474eada6b315